share/wake/lib/system/runner.wake

```diff
@@ -47,6 +47,12 @@ export tuple RunnerInput =
     export Record: Usage
     # Determines if job should run in psuedoterminal
     export IsAtty: Boolean
+    # Modify the Runner's reported inputs (files read).
+    # May only be called once for a given job. Wrapping runners are required to maintain this invariant
```

Comment:
I think this "May" implies it could be run 0 or 1 times, but it really should be called exactly once.

Reply:
That's a curious difference between "may" and "may *only*" -- with the right emphasis, "*may* only" could even be saying that it's expected to be multiple times, but in certain cases it could get as low as one.  Probably why there's an entire RFC dedicated to defining the word for use in specifications.

Reply:
https://datatracker.ietf.org/doc/html/rfc2119

I suggest "Must" but you could use "Shall" if you want some spice in your vocabulary
And maybe include "exactly once"

Reply:
Also, we should probably discuss *why* the invariant's required somewhere, and that the proper way to modify the list when wrapping a runner is to modify the function being passed down.